src/constraints.jl

```diff
@@ -0,0 +1,382 @@
+# type 1
+# ------------------------------------------------------------------------------
+
+const GAE{C,V} = JuMP.GenericAffExpr{C,V}
+const GAEv{C} = JuMP.GenericAffExpr{C,JuMP.VariableRef}
+const GAEp{C} = JuMP.GenericAffExpr{C,Parameter}
+
+mutable struct DoubleGenericAffExpr{C,V,P} <: JuMP.AbstractJuMPScalar
+    v::JuMP.GenericAffExpr{C,V}
+    p::JuMP.GenericAffExpr{C,P}
+end
+const DGAE{C,V,P} = DoubleGenericAffExpr{C,V,P}
+
+const ParametrizedGenericAffExpr{C,V} = DoubleGenericAffExpr{C, V, Parameter}
+const PGAE{C,V} = ParametrizedGenericAffExpr{C,V}

```

Comment:
@jd-lara you might wanna try this type!

Reply:
I will update PowerSimulations to use this now in the ExpressionArrays

Reply:
This is working pretty well!